File: javascript/entire-file.js
const fs = require('fs')

if (process.argv.length !== 3) {
  console.log('Usage: node line-per-line.js <filename>')
  process.exit(1)
}

const data = fs.readFileSync(process.argv[2], 'utf8')

const emailPattern = /[\w\.-]+@[\w\.-]*\.[\w]{2,}/g
const uriPattern = /[\w]+:\/\/[\w\.-]*(?:\/[\w\.-?=&]*)?/g
const ipPattern = /\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/g

let emailCount = 0
let uriCount = 0
let ipCount = 0

let matches = data.match(emailPattern)
if (matches) {
  emailCount += matches.length
}

matches = data.match(uriPattern)
if (matches) {
  uriCount += matches.length
}

matches = data.match(ipPattern)
if (matches) {
  ipCount += matches.length
}

console.log(`${emailCount} emails found.`)
console.log(`${uriCount} URIs found.`)
console.log(`${ipCount} IPs found.`)

File: python/entire-file.py
import sys
import re

if len(sys.argv) != 2:
    print('Usage: python line-per-line.py <filename>')
    sys.exit(1)

emailRegex = re.compile('[\w\.-]+@[\w\.-]*\.[\w]{2,}')
uriRegex = re.compile('[\w]+:\/\/[\w\.-]*(?:\/[\w\.-?=&]*)?')
ipRegex = re.compile('\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}')

emailCount = 0
uriCount = 0
ipCount = 0

with open(sys.argv[1]) as file:
    data = file.read()

    count = len(re.findall(emailRegex, data))
    if count > 0:
        emailCount += count

    count = len(re.findall(uriRegex, data))
    if count > 0:
        uriCount += count

    count = len(re.findall(ipRegex, data))
    if count > 0:
        ipCount += count

print(str(emailCount) + ' emails found.')
print(str(uriCount) + ' URIs found.')
print(str(ipCount) + ' IPs found.')
